digital-land/submit
# Check Tool: Deep links support

## What type of PR is this? (check all applicable)


## Description

- new route: '/check/link' which requires query params and
  redirects to the correct step in the wizard
- added DeepLinkController - to populate the session/history
  with passed query params
- new filter for generating a deep link (path+params)
- removed cookie parser (express docs say it's no longer needed and can
  actually cause issues)
- update check tool templates to display the dataset and organisation
  names when the wizard was entered via a deep link
- some utility functions/maps to simplify looking up datasets info
- moved a predicate out of DatasetController so it can be reused

## Related Tickets & Documents

- Closes #307 

## QA Instructions, Screenshots, Recordings

You can enter the check tool via one of the check service links on a dataset's "Get Started" page, for example: [here](https://submit-pr-550.herokuapp.com/organisations/local-authority:AYL/article-4-direction/get-started). This will take you to the check tool wizard form with some of the data pre-filled (notice it displays the dataset name): 

<img width="651" alt="image" src="https://github.com/user-attachments/assets/719f387d-db00-4855-a03b-98116eae6d80">

Note: clicking "back" to the "choose dataset"  will reset the journey and remove the dataset name from the heading (as if you started the form from the beginning, not through a deep link)

## Added/updated tests?
